### ISSUE TITLE
Testing EspruinoHub

### ISSUE THREAD

**Issue by gcgarner** [gcgarner/IOTstack#84]
Thread for issues of espruinohub

**Comment by sebi5361** [comment]
I didn't try the container yet. However I installed EspruinoHub natively and it works perfectly.
```
git clone https://github.com/espruino/EspruinoHub
# Install EspruinoHub's required Node libraries
cd EspruinoHub
npm install
# Optional - enable gathering of historical data by creating a 'log' directory
mkdir log
# Give Node.js access to Bluetooth
sudo setcap cap_net_raw+eip $(eval readlink -f `which node`)
```
Did you make sure Node.js has access to Bluetooth (last line of the native install procedure)?

**Comment by sebi5361** [comment]
I have noticed that the version of EspruinoHub in the container is really not that stable and loses many connections to BLE devices compared to the latest version when doing a native install.

**Comment by gcgarner** [comment]
Thanks for the feedback, are you running the native and the docker version at the same time?

**Comment by gcgarner** [comment]
The host grants additional rights over just the network port, it seems to allow access to the bluetooth as well. I would rather not have the container operate at host level but that is the recommendations by the creator.

**Comment by gcgarner** [comment]
I had a look at the image and it was updated last 2 years ago

there is an alternative on docker hub hat is 2 months old, will have a look at that one radhus/espruinohub

**Comment by sebi5361** [comment]
> Thanks for the feedback, are you running the native and the docker version at the same time?

No, I uninstalled the latest native version to try the docker container from the stack. Thus I have a good comparison standpoint.
Indeed the two month old radhus version might be a better pick.
I didn't know about host level granting bluetooth access, interesting...
